formula/src/main/java/com/instacart/formula/internal/SnapshotImpl.kt

```diff
@@ -60,6 +60,11 @@ internal class SnapshotImpl<out Input, State> internal constructor(
     }
 
     override fun createScopedKey(type: KClass<*>, key: Any?): Any {
+        if (scopeKey == null && key == null) {
+            // No need to allocate a new object, just use type as key.
+            return type
```

Comment:
Probably worth adding a test for this?

Reply:
There really isn't a great way to test this independently. However, there are a ton of tests within `FormulaRuntimeTest` which would capture if something would break with key management.